base/logging_new_config.go

```diff
@@ -0,0 +1,243 @@
+	logger *log.Logger
+}
+
+type LogFileConfig struct {
+	Enabled  *bool              `json:",omitempty"` // Toggle for this log output
+	Rotation *logRotationConfig `json:",omitempty"` // Log rotation settings
+
+	output io.Writer
+	logger *log.Logger
+}
+
+type logRotationConfig struct {
+	MaxSize   *int `json:",omitempty"` // The maximum size in MB of the log file before it gets rotated.
+	MaxAge    *int `json:",omitempty"` // The maximum number of days to retain old log files.
+	LocalTime bool `json:",omitempty"` // If true, it uses the computer's local time to format the backup timestamp.
+}
+
+func InitLogging(c *LoggingConfig) error {
+	return c.init()
+}
+
+func (c *LoggingConfig) init() error {
+	if c == nil {
+		return errors.New("invalid LoggingConfig")
+	}
+
+	// Use values from the old "default" config for unset values (for backwards compatibility)
+	if c.DeprecatedDefaultLog != nil {
+		if c.LogFilePath == "" && c.DeprecatedDefaultLog.LogFilePath != nil {
+			Warnf(KEY_ALL, "Using deprecated config option: logging.default.logFilePath. Use logging.logFilePath instead.")
+			c.LogFilePath = *c.DeprecatedDefaultLog.LogFilePath
+		}
+		if len(c.Console.LogKeys) == 0 && len(c.DeprecatedDefaultLog.LogKeys) > 0 {
+			Warnf(KEY_ALL, "Using deprecated config option: logging.default.logKeys. Use logging.console.logKeys instead.")
+			c.Console.LogKeys = c.DeprecatedDefaultLog.LogKeys
+		}
+		if c.Console.LogLevel == nil && c.DeprecatedDefaultLog.LogLevel != 0 {
+			Warnf(KEY_ALL, "Using deprecated config option: logging.default.logLevel. Use logging.console.logLevel instead.")
+			newLogLevel := LEVEL_ERROR
+			switch c.DeprecatedDefaultLog.LogLevel {
+			case DebugLevel:
+				newLogLevel.Set(LEVEL_DEBUG)
+			case InfoLevel:
+				newLogLevel.Set(LEVEL_INFO)
+			case WarnLevel:
+				newLogLevel.Set(LEVEL_WARN)
+			}
+			c.Console.LogLevel = &newLogLevel
+		}
+	}
+
+	if err := validateLogFilePath(&c.LogFilePath); err != nil {
+		return err
+	}
+
+	if err := c.Error.init(LEVEL_ERROR, c.LogFilePath, errorMinAge); err != nil {
+		return err
+	}
+	errorLogger = c.Error
+
+	if err := c.Warn.init(LEVEL_WARN, c.LogFilePath, warnMinAge); err != nil {
+		return err
+	}
+	warnLogger = c.Warn
+
+	if err := c.Info.init(LEVEL_INFO, c.LogFilePath, infoMinAge); err != nil {
+		return err
+	}
+	infoLogger = c.Info
+
+	if err := c.Debug.init(LEVEL_DEBUG, c.LogFilePath, debugMinAge); err != nil {
+		return err
+	}
+	debugLogger = c.Debug
+
+	if err := c.Console.init(); err != nil {
+		return err
+	}
+	consoleLogger = c.Console
+
+	return nil
+}
+
+func (lcc *LogConsoleConfig) init() error {
+	if lcc == nil {
+		return errors.New("invalid LogConsoleConfig")
+	}
+
+	// Default to os.Stderr if alternative output is not set
+	if lcc.output == nil {
+		lcc.output = os.Stderr
+	}
+	lcc.logger = log.New(lcc.output, "", 0)
+
+	// Default log level
+	if lcc.LogLevel == nil {
+		newLevel := LEVEL_INFO
+		lcc.LogLevel = &newLevel
+	}
+
+	// HTTP log key is always on
```

Comment:
Previously it was possible to disable HTTP with a call to the _logging endpoint - is that still true?

Reply:
I've overlooked the `_logging` endpoints, but it should still be possible once those methods are tweaked.

I guess this should be done when the old logging code is removed/replaced, rather than when just adding new logging infrastructure.